lib/PrintAsClang/PrintClangFunction.cpp

```diff
@@ -1238,13 +1241,49 @@ void DeclAndTypeClangFunctionPrinter::printCxxThunkBody(
   // Create the condition and the statement to throw an exception.
   if (hasThrows) {
     os << "  if (opaqueError != nullptr)\n";
+    os << "#ifdef __cpp_exceptions\n";
     os << "    throw (Swift::Error(opaqueError));\n";
+    os << "#else\n";
+    if (resultTy->isVoid()) {
+      os << "    return SWIFT_RETURN_THUNK(void, Swift::Error(opaqueError));\n";
+    } else {
+      os << "    return SWIFT_RETURN_THUNK(";
+      auto directResultType = signature.getDirectResultType();
+      printDirectReturnOrParamCType(
```

Comment:
Maybe we can extract the return type printing code into a closure in this function, to avoid repeating this code?